src/azure-client.ts

```diff
@@ -0,0 +1,48 @@
+import * as fs from "fs";
+
+import FormData from "form-data";
+
+import { getApiClient } from "./gh-api-client";
+
+export interface Policy {
+  upload_url: string;
+  header: Map<string, string>;
+  form: Map<string, string>;
```

Comment:
We might have gone a bit overboard by declaring these as a `Map`. I wasn't sure if `Map` was just a synonym for `Object` but it is not. They're different and not really interchangeable. See https://medium.com/dailyjs/7-differences-between-objects-and-maps-in-javascript-bc901dfa9350 for some more info.

The particularly problematic bits for us are that:
- When you deserialise some JSON it'll always become an object. Ideally we'd like to able to just `JSON.parse` the result of the API request to get the policy.
- Later on when we do `...policy.header`, that doesn't work with a `Map`. It's not a compile error but it just does nothing.

Luckily we can make these fields objects but still specify types for them. I'm my opinion the syntax is a bit arcane and less readable, but it works. I suggest we replace these two lines with:
```suggestion
  header: [name: string]: string;
  form: [name: string]: string;
```
So this is still saying that our object has keys of type `string` and values of type `string`.

Reply:
Instead of `[name: string]: string`, you can also use `Record<string, string>` which is equivalent but might be a bit easier to read.

Reply:
Awesome, thanks both 😍 🙌🏽